react/Modal/index.jsx

```diff
@@ -9,9 +9,94 @@ import migrateProps from '../helpers/migrateProps'
 import palette from '../../stylus/settings/palette.json'
 import Portal from 'preact-portal'
 
-const ModalContent = ({ children, className }) => (
-  <div className={cx(styles['c-modal-content'], className)}>{children}</div>
-)
+class ModalContent extends Component {
+  constructor(props) {
+    super(props)
+    this.state = { displayGhostHeader: false }
+  }
+
+  componentDidMount() {
+    this.scrollingContent.addEventListener('scroll', this.handleScroll, {
+      passive: true
+    })
+  }
+
+  handleScroll = () => {
+    if (
+      !this.state.displayGhostHeader &&
+      this.scrollingContent.scrollTop > 56
+    ) {
+      this.setState(() => ({ displayGhostHeader: true }))
+    } else if (
+      this.state.displayGhostHeader &&
+      this.scrollingContent.scrollTop < 56
+    ) {
+      this.setState(() => ({ displayGhostHeader: false }))
+    }
+  }
+
+  render() {
+    const { iconSrc, iconDest, service, children, className } = this.props
+
+    const { displayGhostHeader } = this.state
+
+    return (
+      <div
+        className={cx(styles['c-modal-content'], className)}
+        ref={div => {
+          this.scrollingContent = div
+        }}
+      >
+        {iconSrc &&
+          iconDest && (
+            <div>
+              <h2 className={cx(styles['c-modal-header--icon'], className)}>
+                <div>
```

Comment:
Why this extra div ?

Reply:
Because if I don't add it, I get the error `Syntax error: Adjacent JSX elements must be wrapped in an enclosing tag`

Reply:
ah no sorry, not this one, the other one inside. You're right, that's a leftover from a previous attempt. It is indeed useless.